### PR TITLE
Moved and reorged remaining api convention pages

### PR DESCRIPTION
**Pull request by ryangyoung**

### CODE REVIEW

File: content/references/rippled-api/api-conventions/basic-data-types.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+[Accounts](concept-accounts.html) are identified by their [Address][], for example `"r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59"`. Addresses always start with "r". Many `rippled` methods also accept a hexadecimal representation.
+
+[Transactions](reference-transaction-format.html) are identified by a [Hash][] of the transaction's binary format. You can also identify a transaction by its sending account and [Sequence Number][].
+
+Each closed [Ledger](reference-ledger-format.html) has a [Ledger Index][] and a [Hash][] value. When [Specifying a Ledger Instance](#specifying-ledgers) you can use either one.
+
+## Addresses
+[Address]: #addresses
+
+{% include '_snippets/data_types/address.md' %}
```
Comment (mDuo13) [inline content/references/rippled-api/api-conventions/basic-data-types.md:14]: I am really starting to regret going with this convention of starting folder names with underscores for things that aren't included in the docs directly... 🙀

File: dactyl-config.yml [hunk 2]
```diff
@@ -459,7 +459,61 @@ pages:
         targets:
             - local
 
-    # TODO: rippled Server States, Error Formatting
+    -  md: references/rippled-api/api-conventions/basic-data-types.md
```
Comment (mDuo13) [inline dactyl-config.yml]: "Basic Data Types" should come before "Currency Formatting" (see line 451)

File: content/references/rippled-api/api-conventions/basic-data-types.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    * `current` for the server's current working version of the ledger.
+
+There is also a deprecated `ledger` parameter which accepts any of the above three formats. *Do not* use this parameter; it may be removed without further notice.
+
+If you do not specify a ledger, the `current` (in-progress) ledger is chosen by default. If you provide more than one field specifying ledgers, the deprecated `ledger` field is used first if it exists, falling back to `ledger_hash`. The `ledger_index` field is ignored unless neither of the other two are present.
+
+**Note:** Do not rely on this default behavior for specifying a ledger; it is subject to change. Always specify a ledger version in the request if you can.
+
+
+## Currencies
```
Comment (mDuo13) [inline content/references/rippled-api/api-conventions/basic-data-types.md:53]: Can we find a way to split this off and merge it with the "Currency Formatting" page? I believe the two areas have a lot of redundant content.

**Review by mDuo13 — CHANGES_REQUESTED** [review]
Looks pretty good. I'd like to see the "Basic Data Types" page put in the correct order, and I'm not sure whether there's time to properly synthesize the currency formatting overlap but if so that would be nice to do too.

Another (optional) suggestion: if you add a blurb to each page in the API Conventions category, then the child listing on the landing page can use those to provide more context to the links.

Also add a blurb to the API Conventions landing page itself to appear on the "rippled API" landing page.

File: content/references/rippled-api/api-conventions/basic-data-types.md (hunk 3 above)
Comment (mDuo13) [reply content/references/rippled-api/api-conventions/basic-data-types.md:53]: Maybe this is too much work and we should just add a task to the backlog to combine them.

File: content/references/rippled-api/api-conventions/basic-data-types.md (hunk 1 above)
Comment (ryangyoung) [reply content/references/rippled-api/api-conventions/basic-data-types.md:14]: Yeah, a lot of purple text in Atom...

File: content/references/rippled-api/api-conventions/basic-data-types.md (hunk 3 above)
Comment (ryangyoung) [reply content/references/rippled-api/api-conventions/basic-data-types.md:53]: Created https://ripplelabs.atlassian.net/browse/DOC-1549 to track that

**Comment by ryangyoung** [comment]
@mDuo13 created https://ripplelabs.atlassian.net/browse/DOC-1550 for adding the blurbs

**Review by mDuo13 — APPROVED** [review]

**Pull request merged**